index.js

```diff
@@ -1,14 +1,16 @@
 'use strict';
-module.exports = function (a, b) {
+module.exports = function (a, b, minComp) {
 	if (!Buffer.isBuffer(a) || !Buffer.isBuffer(b)) {
 		throw new TypeError('Arguments must be Buffers');
 	}
 
+	var len = Math.max(a.length, b.length, minComp || 0);
```

Comment:
About `a.length`:

An implementer may use `bufferEqualsConstant(a, b)` or `bufferEqualsConstant(b, a)`, so the problem remains. Even more I think `bufferEqualsConstant` is not constant because:

```
time(bufferEqualsConstant(a, b)) != time(bufferEqualsConstant(b, a))
```